lib/routes/token.js

```diff
@@ -236,6 +240,32 @@ module.exports = {
       }
       return vals;
     })
+    .then(function(vals) {
+      // Incremental authorization.
+      if (params.existing_grant) {
+        return db.getRefreshToken(encrypt.hash(params.existing_grant)).then(function (tok) {
+          if (!tok) {
+            throw new AppError.unauthorized('Unknown existing grant');
+          }
+          if (tok.userId.toString('hex') !== vals.userId.toString('hex') ||
+              tok.clientId.toString('hex') !== vals.clientId.toString('hex')) {
+            throw new AppError.unauthorized('Invalid grant');
+          }
+          vals.scope = Scope(vals.scope).merge(Scope(tok.scope)).values();
+          return vals;
+        });
+      } else if (vals.includeGrantedScopes) {
+        return db.getActiveClientTokensByUid(vals.userId.toString('hex')).then(function (tokens) {
+          const allScopes = []; // Contain duplicates, but Scope#merge drops them.
+          for (const token of tokens) {
+            allScopes.push(...token.scope);
```

Comment:
Seems strange to me that the singular `scope` is really an array, but that's not an issue caused by this PR.